Dockerfile

```diff
@@ -0,0 +1,140 @@
+# All build arguments
+
+# TARGETPLATFORM - automatic assignment via `docker build --platform xyz`
+# RUST_PROFILE - defaults to release
+
+# Build babylon-node binary
+FROM debian:11-slim AS binary-build-stage
+LABEL org.opencontainers.image.authors="devops@radixdlt.com"
+LABEL org.opencontainers.image.description="Java + Debian 11 (OpenJDK)"
+
+ENV DEBIAN_FRONTEND noninteractive
+
+CMD ["/bin/bash"]
+
+RUN  \
+  apt-get update && apt-get install -y --no-install-recommends \
+  docker.io=20.10.5+dfsg1-1+deb11u2 \
+  libssl-dev=1.1.1n-0+deb11u4 \
+  pkg-config=0.29.2-1 \
+  unzip=6.0-26+deb11u1 \
+  wget=1.21-1+deb11u1 \
+  software-properties-common=0.96.20.2-2.1 
+RUN add-apt-repository -y ppa:openjdk-r/ppa && \
+  apt-get install -y --no-install-recommends \
+  openjdk-17-jdk=17.0.6+10-1~deb11u1
+
+#install Gradle
+RUN wget -q https://services.gradle.org/distributions/gradle-7.2-bin.zip \
+  && unzip gradle-7.2-bin.zip -d /opt \
+  && rm gradle-7.2-bin.zip
+
+# Set Gradle in the environment variables
+ENV GRADLE_HOME=/opt/gradle-7.2
+ENV PATH=/opt/gradle-7.2/bin:$PATH
+ENV JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
+
+COPY . /radixdlt
+WORKDIR /radixdlt
+USER root
+RUN SKIP_NATIVE_RUST_BUILD=TRUE gradle clean build -x test -Pci=true -PrustBinaryBuildType=release
+
+USER nobody
+
+# Exporting babylon-node binary
+FROM scratch AS binary-container
+COPY --from=binary-build-stage /radixdlt/core/build/distributions /
+
+# Building core-rust library
+FROM lukemathwalker/cargo-chef:0.1.52-rust-1.68.0 AS chef
+LABEL org.opencontainers.image.authors="devops@radixdlt.com"
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    clang=1:11.0-51+nmu5 \
+    libmpc-dev=1.2.0-1 \
+    libmpfr-dev=4.1.0-3 \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+WORKDIR /app
+WORKDIR /app
+
+FROM chef AS planner
+COPY core-rust ./
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS library-builder 
+ENV RUST_BACKTRACE=1
+RUN apt-get update && apt-get install libclang-dev=1:11.0-51+nmu5 -y --no-install-recommends
+COPY --from=planner /app/recipe.json recipe.json
+
+ARG TARGETPLATFORM
+ARG RUST_PROFILE=release
+COPY docker/scripts/cargo_build_by_platform.sh /opt/radixdlt/cargo_build_by_platform.sh
+RUN /opt/radixdlt/cargo_build_by_platform.sh chef $TARGETPLATFORM release
+# Build application
+COPY core-rust ./
+RUN /opt/radixdlt/cargo_build_by_platform.sh build $TARGETPLATFORM release
+
+# Exporting core-rust library
+FROM scratch as library-container
+COPY --from=library-builder /libcorerust.so /
+
+# Building the application container
+FROM debian:11-slim as app-container
+LABEL org.opencontainers.image.authors="devops@radixdlt.com"
+
+RUN apt-get update -y && \
+    apt-get -y --no-install-recommends install \
+    openjdk-17-jre-headless=17.0.6+10-1~deb11u1 \
+    unzip=6.0-26+deb11u1 \
+    daemontools=1:0.76-7 \
+    libssl-dev=1.1.1n-0+deb11u4 \
+    software-properties-common=0.96.20.2-2.1 \
+    gettext-base=0.21-4 && \
```

Comment:
Have you worked out why we need these dependencies? If so, can we add comments for why we need them? If not, no worries

> daemontools=1:0.76-7 \
    libssl-dev=1.1.1n-0+deb11u4 \
    software-properties-common=0.96.20.2-2.1 \
    gettext-base